### PR TITLE
Improved Invoice list loading speed, and fields displayed.

### PR DESCRIPTION
**Pull request by juanvallejo-eb**
The invoice list no longer loads the invoice file, now it's only
available in the detail view.

Also fixed the invoice creation being able to re-submit the same data
multiple times.

### CODE REVIEW

File: templates/supplier_app/invoices_form.html [hunk 1]
```diff
@@ -107,4 +107,13 @@ <h4 class="text-center">New Invoice</h4>
     </div>
 </div>
 
+</script>
+<script type="text/javascript">
+$(document).ready(function () {
```
Comment (pablogumilla) [inline templates/supplier_app/invoices_form.html]: Si querés, podés usar una forma más corta para escribir `$(document).ready(function()` al usar jQuery, que es la siguiente:

```
$(function() {
...
};
```

File: invoices_app/forms.py [hunk 2]
```diff
@@ -51,5 +51,14 @@ def is_valid(self):
...
             return False
-
-        return True
+        invoice_number = self.cleaned_data['invoice_number']
+        taxpayer = self.data['taxpayer']
+        try:
+            Invoice.objects.get(taxpayer=taxpayer, invoice_number=invoice_number)
+            self.add_error(
+                'invoice_number',
+                'The invoice {} already exists'.format(invoice_number)
```
Comment (martinvalles-eb) [inline invoices_app/forms.py]: Excellent!

File: invoices_app/views.py [hunk 3]
```diff
@@ -48,9 +48,9 @@ def get_context_data(self, **kwargs):
     def get_queryset(self):
         user = self.request.user
         if user.is_AP:
-            queryset = Invoice.objects.filter(status=INVOICE_STATUS_NEW).order_by('id')
+            queryset = Invoice.objects.filter(status=INVOICE_STATUS_NEW).defer('invoice_file').order_by('id')
         else:
-            queryset = Invoice.objects.filter(user=user)
+            queryset = Invoice.objects.filter(user=user).defer('invoice_file')
```
Comment (martinvalles-eb) [inline invoices_app/views.py]: Good call to use `defer` here

File: templates/invoices_app/invoice_detail.html [hunk 4]
```diff
@@ -87,6 +87,7 @@ <h6 class="mb-3">Details:</h6>
     <span><button type="button" class="btn btn-info">Edit info</button></span>
     {% endif %}
     {% if is_AP %}
+    <span><a href="{% url 'invoice-history' pk=invoice.id %}" ><button type="button" class="btn btn-primary">History</button></a></span>
```
Comment (martinvalles-eb) [inline templates/invoices_app/invoice_detail.html]: Why do you guys use this html structure here?
`<a><button>...</button></a>`

In terms of semantic, anchor tags(<a>) are used for links that redirects you to another page and button(<button>) are used to trigger actions or functions o some sort of "behavior".

File: templates/invoices_app/invoice_detail.html [hunk 5]
```diff
@@ -83,15 +83,18 @@ <h6 class="mb-3">Details:</h6>
...
-    {% if invoice.status == 'CHANGES REQUEST' %}
-    <span><button type="button" class="btn btn-info">Edit info</button></span>
+    {% if invoice.status == 'CHANGES REQUEST' and not is_AP %}
+      <span><a href="{% url 'taxpayer-invoice-update' taxpayer_id=taxpayer.id pk=invoice.id %}" class="btn btn-info">Edit info</a></span>
     {% endif %}
     {% if is_AP %}
-    <span><a href="{% url 'invoice-approve' pk=invoice.id %}" ><button type="button" class="btn btn-success">Approve</button></a></span>
-    <span><a href="{% url 'invoice-reject' pk=invoice.id %}" ><button type="button" class="btn btn-danger">Reject</button></a></span>
-    <span><button type="button" class="btn btn-warning">Request Changes</button></span>
+    <span><a href="{% url 'invoice-history' pk=invoice.id %}" class="btn btn-primary">History</a></span>
```
Comment (martinvalles-eb) [inline templates/invoices_app/invoice_detail.html]: Much better and cleaner. Good job!

**Pull request merged**